### PR TITLE
feat(http): Cleanup for HTTP cache

### LINKED ISSUE #28376: http cache TTL expiry
### Describe the proposed change(s).

Add a new global option httpCacheTtlDays defaulting to 90

### PR DESCRIPTION
**Pull request by zharinov**
## Changes

- Introduce `httpCacheTtlDays` option
- After each run, check for stale items and remove them
- When `httpCacheTtlDays` is `0`, remove all cache items

## Context

- Closes: #28376

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/config/options/index.ts [hunk 1]
```diff
@@ -2981,6 +2981,15 @@ const options: RenovateOptions[] = [
     default: null,
     supportedPlatforms: ['github'],
   },
+  {
+    name: 'httpCacheTtlDays',
+    description:
+      'Maximum duration in days to keep HTTP cache entries. Set to `0` to disable HTTP cache.',
+    type: 'integer',
+    stage: 'repository',
+    default: 90 * 24 * 60,
```
Comment (rarkins) [inline lib/config/options/index.ts:2990]:
```suggestion
    default: 90,
```

File: lib/util/cache/repository/http-cache.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import is from '@sindresorhus/is';
+import { DateTime } from 'luxon';
+import { GlobalConfig } from '../../../config/global';
+import { logger } from '../../../logger';
+import { HttpCacheSchema } from '../../http/cache/schema';
+
+export function cleanupHttpCache(cacheData: unknown): void {
+  if (!is.plainObject(cacheData) || !is.plainObject(cacheData['httpCache'])) {
+    logger.trace('cleanupHttpCache: invalid cache data');
```
Comment (rarkins) [inline lib/util/cache/repository/http-cache.ts:9]:
```suggestion
    logger.warn('cleanupHttpCache: invalid cache data');
```

Isn't this a rare and unlikely event? (if you exclude empty/null cacheData)

**Review by rarkins — CHANGES_REQUESTED** [review]

**Review by viceice — DISMISSED** [review]

**Review by viceice — DISMISSED** [review]

File: lib/config/options/index.ts [hunk 3]
```diff
@@ -2981,6 +2981,15 @@ const options: RenovateOptions[] = [
     default: null,
     supportedPlatforms: ['github'],
   },
+  {
+    name: 'httpCacheTtlDays',
+    description:
+      'Maximum duration in days to keep HTTP cache entries. Set to `0` to disable HTTP cache.',
```
Comment (rarkins) [inline lib/config/options/index.ts:2987]:
```suggestion
      'Maximum duration in days to keep HTTP cache entries.',
```

Comment (rarkins) [reply lib/config/options/index.ts:2987]: Details are already in the markdown

**Review by rarkins — CHANGES_REQUESTED** [review]

**Review by rarkins — DISMISSED** [review]

**Review by viceice — APPROVED** [review]

**Comment by mjnowen** [comment]
I believe this PR has introduced a bug. I have a Renovate MR in self-hosted GitLab with a >90 day http cache. Instead of it rotating (purging) my cache by default, it just prints a WARN message and returns.
[ cleanupHttpCache](https://github.com/renovatebot/renovate/blob/main/lib/util/cache/repository/http-cache.ts#L9)

I would expect >90 day http cache (by default=90) to not just `return` but actually purge my cache, but as the code is currently written, I never get to this line: https://github.com/renovatebot/renovate/blob/main/lib/util/cache/repository/http-cache.ts#L9

**Pull request merged**